crates/ln-dlc-node/src/ln/app_event_handler.rs

```diff
@@ -234,12 +251,66 @@ where
                 payment_hash,
                 onion_fields: _,
                 amount_msat,
-                counterparty_skimmed_fee_msat: _,
+                counterparty_skimmed_fee_msat,
                 purpose,
                 via_channel_id: _,
-                via_user_channel_id: _,
+                via_user_channel_id,
                 claim_deadline: _,
             } => {
+                if counterparty_skimmed_fee_msat > 0 {
+                    tracing::info!("Checking if counterparty skimmed fee msat is justified");
+                    let user_channel_id = via_user_channel_id.context("Missing user channel id")?;
+                    let user_channel_id = UserChannelId::from(user_channel_id);
+                    let channel = self
+                        .node
+                        .storage
+                        .get_channel(&user_channel_id.to_string())?
+                        .with_context(|| {
+                            format!("Couldn't find channel for user_channel_id {user_channel_id}",)
+                        })?;
+
+                    if channel.channel_state == ChannelState::Open {
+                        tracing::error!(
+                            "Channel opening fees have already been paid. Rejecting payment."
+                        );
+                        self.node.channel_manager.fail_htlc_backwards_with_reason(
+                            &payment_hash,
+                            FailureCode::IncorrectOrUnknownPaymentDetails,
+                        );
+                        return Ok(());
+                    }
+
+                    let channel_fee_msats = channel.fee_sats.unwrap_or(0) * 1000;
+                    if channel_fee_msats < counterparty_skimmed_fee_msat {
+                        tracing::error!("Counterparty skimmed too much fee. Expected fee {channel_fee_msats} msats, Got fee {counterparty_skimmed_fee_msat} msats Rejecting payment!");
+                        self.node.channel_manager.fail_htlc_backwards_with_reason(
+                            &payment_hash,
+                            FailureCode::IncorrectOrUnknownPaymentDetails,
+                        );
+                        return Ok(());
+                    }
+
+                    tracing::info!(
+                        funding_txid=?channel.funding_txid,
+                        "Paying channel opening fee of {}",
+                        counterparty_skimmed_fee_msat / 1000
+                    );
```

Comment:
🙃 I would reword this because it seems to indicate that we are making a payment, but we are really accepting a payment with a reduced amount.